src/server.h

```diff
@@ -148,9 +148,13 @@ typedef long long ustime_t; /* microsecond time type. */
 /* Instantaneous metrics tracking. */
 #define STATS_METRIC_SAMPLES 16     /* Number of samples per metric. */
 #define STATS_METRIC_COMMAND 0      /* Number of commands executed. */
-#define STATS_METRIC_NET_INPUT 1    /* Bytes read to network .*/
+#define STATS_METRIC_NET_INPUT 1    /* Bytes read to network. */
 #define STATS_METRIC_NET_OUTPUT 2   /* Bytes written to network. */
-#define STATS_METRIC_COUNT 3
+#define STATS_METRIC_NET_INPUT_SYNC_FULLRESYNC 3    /* Bytes read from network during replication for fullresync. */
+#define STATS_METRIC_NET_OUTPUT_SYNC_FULLRESYNC 4   /* Bytes written to network during replication for fullresync. */
+#define STATS_METRIC_NET_INPUT_SYNC_CMDS 5    /* Bytes read from network during replication for live command stream. */
+#define STATS_METRIC_NET_OUTPUT_SYNC_CMDS 6   /* Bytes written to network during replication for live command stream. */
```

Comment:
i'm quite certain we don't want instantaneous metrics for all of these.
not for full-sync vs command-stream, and not even sure we need a separate replication input vs replication output.
i.e. these INFO fields are mainly for redis-cli users, not machines, and if they see high output traffic, and they know it's a master, they can deduct the replication bytes.
similarly if they see high input traffic on a replica, they'll deduct the replication bytes from that.

Reply:
If we only use single variable like `net_repl_bytes`, I think it might be ambiguous for structure like 'master->slave->slave'. 
The middle redis node could both be a replica and master at the meantime.

Reply:
yes, but for the "instantaneous" metrics, i'm not sure it's that important.
they're meant for quick manual check via redis-cli, since a proper metrics system will take the raw counters and plot their changes over time.

Reply:
i see you removed all 4 instantaneous metrics you added. i think maybe we can keep one.
i.e. one that's about replication (sum of input and output)

Reply:
@oranagra I think it's better to track and display the instantaneous replication input and output separately, we built read-write split structure as @DarrenJiang13 said: `master->replica->replica`, the middle replica has both input and output replication flow, and it's important to tell users how many input/output flow comes from themselves and others are replication flow.

So we need `instantaneous_input_kbps - instantaneous_repl_input_kbps` and `instantaneous_output_kbps - instantaneous_repl_output_kbps`

Reply:
@soloestoy the way i see it (please tell me if you see it differently) these instantaneous metrics are just for human eyes.
real monitoring systems, take a snapshot of the INFO every second, and can measure the increase of the raw metrics (non-instantaneous) over time.
so for instance if you have a grafana dashboard showing redis metrics, it'll still be able to show the difference between input and output, and it'll also be accurate, since it doesn't depend on these (inaccurate) "instantaneous" metrics (corresponding to 1.6 seconds or so).

so considering that these metrics are just for a person taking one snapshot of INFO via redis-cli to examine it in the console, do we really need another metric for this case?